shared-lib/azure/azure-http/src/main/java/org/opennms/horizon/shared/azure/http/AzureHttpClient.java

```diff
@@ -232,13 +279,42 @@ private <T> T performRequest(String endpoint, Class<T> clazz, HttpRequest reques
                 }
             }
         }
-        Throwable cause = exception.getCause();
-        if (cause instanceof InterruptedException) {
-            Thread.currentThread().interrupt();
+        throw lastException;
+    }
+
+    private <T> T performSingleRequest(Class<T> clazz, HttpRequest request) throws AzureHttpException{
+        try {
+            HttpResponse<String> httpResponse = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+            String httpBody = httpResponse.body();
+
+            log.info("============================");
+            log.info("request: {}", request);
+            log.info("header: {}", request.headers());
+            log.info("statusCode: {}", httpResponse.statusCode());
+            log.info("httpBody: {}", httpBody);
+            log.info("============================");
```

Comment:
I don't think we want to log all of this. Logging the request might expose the client secret/token.

```suggestion
            log.info("Response statusCode: {}, body: {}", httpResponse.statusCode(), httpBody);
```